azure/meta.go

```diff
@@ -0,0 +1,131 @@
+package azure
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hairyhenderson/gomplate/v4/env"
+)
+
+// DefaultEndpoint is the IMDS endpoint at Azure
+// https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service
+var DefaultEndpoint = "http://169.254.169.254"
+
+var (
+	// co is a ClientOptions populated from the environment.
+	co ClientOptions
+	// coInit ensures that `co` is only set once.
+	coInit sync.Once
+)
+
+// ClientOptions contains various user-specifiable options for a MetaClient.
+type ClientOptions struct {
+	Timeout time.Duration
+}
+
+// GetClientOptions - Centralised reading of AZURE_TIMEOUT
+// ... but cannot use in vault/auth.go as different strconv.Atoi error handling
+func GetClientOptions() ClientOptions {
+	coInit.Do(func() {
```

Comment:
Can you figure out why there's a `sync.Once` involved here? I've been digging through past history, and the pattern seems to come from https://github.com/hairyhenderson/gomplate/pull/248, though I honestly can't really tell _why_...

I think I'd rather remove the `coInit`, and return a `(ClientOptions, error)`, and error on `strconv.Atoi` rather than panic.